conda_index/cli/__init__.py

```diff
@@ -91,6 +92,24 @@
         repodata_version=2 which is supported in conda 24.5.0 or later.
         """,
 )
+@click.option(
+    "--update-cache/--no-update-cache",
+    help="""
+        Control whether listdir() is called to refresh the set of available
+        packages. Used to generate complete repodata.json from cache only when
+        packages are not on disk. (Experimental)
+        """,
+    default=True,
+    show_default=True,
+)
+@click.option(
+    "--upstream-stage",
```

Comment:
The stat table has multiple rows per package and stage. If a package only exists in stage=`fs` or `clone` then its metadata is cached and a `stage=index` row is added. This option doesn't explain the mechanism.